src/rules/no-expect-to-match-object.ts

```diff
@@ -0,0 +1,36 @@
+import { createRule, isExpectCall, parseExpectCall } from './utils';
+
+export default createRule({
+  name: __filename,
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description: 'Disallow expect.toMatchObject',
+      recommended: false,
+    },
+    messages: {
+      useToStrictEqual: 'Use toStrictEqual() for a deep equality check',
```

Comment:
should this include something about `expect.objectContaining` as well?

Reply:
Yeah I'm thinking maybe this rule should be `prefer-object-containing` and push that, rather than pushing for `toStrictEqual` since that's covered by `prefer-strict-equal` where as you can use `expect.objectContaining` with other matches.

(I can't remember off the top of my head if `toBe` with `expect.objectContaining` behaviours any similar to `toStrictEqual`)

Reply:
`toBe` wouldn't make sense as that's for referential equality

Reply:
> should this include something about `expect.objectContaining` as well?
> @SimenB 

This was like this in the first commit of this PR: https://github.com/jest-community/eslint-plugin-jest/blob/5ecba757567d02f0a618fb6c12fbba7074d1b79e/src/rules/no-expect-to-match-object.ts#L17-L20
Should it be something like it then? , how about ?:

`Use toStrictEqual() for a deep equality check, use expect.objectContaining() to check specific properties`

Reply:
> Yeah I'm thinking maybe this rule should be `prefer-object-containing` and push that, rather than pushing for `toStrictEqual` since that's covered by `prefer-strict-equal` where as you can use `expect.objectContaining` with other matches.
> @G-Rath 

`prefer-strict-equal` does not prevent the use of `expect().toMatchObject(object)`

Reply:
Correct - my point was about that you can use `expect.objectContaining` with matchers other than `toStrictEqual` (`toEqual` for example, which would have the same result in most cases).

Looking back over this, I think maybe this is just another one for a general "no-banned-matches" rule since the matcher does have uses, and doesn't have any particular conditions around it like `prefer-to-contain`, `prefer-to-have-length`, etc... 🤔 

If I get the time this weekend I might try implementing such a rule.